packages/tools/src/utilities/segmentation/getSegmentAtWorldPoint.ts

```diff
@@ -0,0 +1,146 @@
+import { cache, utilities } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { SegmentationRepresentations } from '../../enums';
+import {
+  findSegmentationRepresentationByUID,
+  getSegmentation,
+} from '../../stateManagement/segmentation/segmentationState';
+import {
+  LabelmapSegmentationData,
+  LabelmapSegmentationDataStack,
+  LabelmapSegmentationDataVolume,
+} from '../../types/LabelmapTypes';
+import { isVolumeSegmentation } from '../../tools/segmentation/strategies/utils/stackVolumeCheck';
+import {
+  ContourSegmentationAnnotation,
+  ContourSegmentationData,
+} from '../../types';
+import { getAnnotation } from '../../stateManagement';
+import { isPointInsidePolyline3D } from '../math/polyline';
+
+/**
+ * Get the segment at the specified world point in the viewport.
+ * @param viewport - The viewport where the point resides.
+ * @param worldPoint - The world point to get the segment for.
+ * @param segmentationId - The ID of the segmentation to lookup.
+ * @param representationType - The type of segmentation representation (labelmap, contour etc.)
+ * @returns The index of the segment at the world point, or undefined if not found.
```

Comment:
What happens if there are multiple contours at the same location?  I know for our current labelmaps, we don't allow that, but do you need to support it for the general case?

Reply:
It has the segmentationRepresentationUID so that would not be an issue